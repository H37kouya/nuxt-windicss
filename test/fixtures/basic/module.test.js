import { setupTest, getNuxt } from '@nuxt/test-utils'

const cheerio = require('cheerio')

describe('Dev test', () => {

  // @ts-ignore
  setupTest({
    testDir: __dirname,
    fixture: __dirname,
    configFile: 'nuxt.config.ts',
    build: true,
    config: {
      dev: true,
    },
  })

  test('renders index route', async() => {
    // @ts-ignore
    const { html } = await getNuxt().server.renderRoute('/')
    const $ = cheerio.load(html)
    expect($('style').first().html()).toMatchSnapshot()
  })

})

describe('Production test', () => {
  // @ts-ignore
  setupTest({
    testDir: __dirname,
    fixture: __dirname,
    configFile: 'nuxt.config.ts',
    build: true,
    config: {
      dev: false,
    },
  })

  test('renders index route', async() => {
    // @ts-ignore
    const { html } = await getNuxt().server.renderRoute('/')
    const $ = cheerio.load(html)
    expect($('style').first().html()).toMatchSnapshot()
  })
})
