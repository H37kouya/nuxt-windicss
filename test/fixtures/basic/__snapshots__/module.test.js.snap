// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dev test renders index route 1`] = `
".scss-global {
  --tw-bg-opacity: 1;
  background-color: rgba(52, 211, 153, var(--tw-bg-opacity));
  --tw-text-opacity: 1;
  color: rgba(255, 255, 255, var(--tw-text-opacity));
  padding: 1rem;
  width: 25%;
  transition-property: background-color, border-color, color, fill, stroke, opacity, box-shadow, transform;
  transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
  transition-duration: 150ms;
}

.scss-global:hover {
  --tw-bg-opacity: 1;
  background-color: rgba(6, 78, 59, var(--tw-bg-opacity));
  --tw-text-opacity: 1;
  color: rgba(224, 242, 254, var(--tw-text-opacity));
}

.scss-global h2 {
  font-weight: 700;
  font-size: 0.875rem;
  line-height: 1.25rem;
}"
`;

exports[`Production test renders index route 1`] = `"h1{--tw-bg-opacity:1;background-color:rgba(7,89,133,var(--tw-bg-opacity))}h1 span{--tw-text-opacity:1;color:rgba(52,211,153,var(--tw-text-opacity))}"`;
